File: sdk/java-sdk/src/main/java/com/evernym/verity/sdk/utils/JsonUtil.java
package com.evernym.verity.sdk.utils;

import org.json.JSONArray;

import java.util.Arrays;
import java.util.List;

/**
 * Utilities for creating JSON objects
 */
public class JsonUtil {
    private JsonUtil(){}

    /**
     * Builds a JSONArray from an array of object that implement AsJsonObject interface
     * @param items an array of object that can be converted to an JSONObject
     * @return a JSON array
     */
    public static JSONArray makeArray(AsJsonObject[] items) {
        return buildArray(Arrays.asList(items));
    }

    /**
     * Builds a JSONArray from a list of object that implement AsJsonObject interface
     * @param items a List of object that can be converted to an JSONObject
     * @return a JSON array
     */
    public static JSONArray makeArrayFromList(List<AsJsonObject> items) {
        return buildArray(items);
    }

    private static JSONArray buildArray(Iterable<AsJsonObject> items) {
        JSONArray rtn = new JSONArray();
        if(items != null) {
            for(AsJsonObject i: items) {
                if(i != null) {
                    rtn.put(i.toJson());
                }
            }
        }
        return rtn;
    }
}

File: samples/sdk/dotnet-example-app-tmp/VeritySDK/Utils/JsonUtil.cs
using System;
using System.Collections.Generic;
using System.Json;
using System.Linq;
using System.Text;

namespace VeritySDK.Utils
{
    /// <summary>
    /// Utilities for creating JSON objects
    /// </summary>
    public class JsonUtil
    {
        private JsonUtil() { }

        /// <summary>
        /// Builds a JsonArray from an array of object that implement AsJsonObject interface 
        /// </summary>
        /// <param name="items">items an array of object that can be converted to an JsonObject</param>
        /// <returns>a JSON array</returns>
        public static JsonArray makeArray(AsJsonObject[] items)
        {
            return buildArray(items.AsEnumerable());
        }

        /// <summary>
        /// Builds a JSONArray from a list of object that implement AsJsonObject interface 
        /// </summary>
        /// <param name="items">a List of object that can be converted to an JsonObject</param>
        /// <returns>a JSON array</returns>
        public static JsonArray makeArrayFromList(List<AsJsonObject> items)
        {
            return buildArray(items);
        }

        private static JsonArray buildArray(IEnumerable<AsJsonObject> items)
        {
            JsonArray rtn = new JsonArray();
            if (items != null)
            {
                foreach (var i in items)
                {
                    if (i != null)
                    {
                        rtn.Add(i.toJson());
                    }
                }
            }
            return rtn;
        }
    }
}
